modules/flann/include/opencv2/flann/flann_base.hpp

```diff
@@ -45,6 +45,37 @@
 
 namespace cvflann
 {
+class FILEScopeGuard {
+
+public:
+    FILEScopeGuard& operator=(FILEScopeGuard&&) = default;
+    FILEScopeGuard() = default;
+    FILEScopeGuard(FILEScopeGuard&&) = default;
+
+    ~FILEScopeGuard() {
+        fclose(file_);
+        std::cout << "FILEScopeGuard: file is closed\n";
+    };
+
+    FILEScopeGuard(FILE* file)
+        : file_{ file }
+    { }
+
+    FILEScopeGuard fopen(const char * filename, char const* mode) {
+        FILE * file = ::fopen(filename, mode);
+        return FILEScopeGuard(file);
+    }
+
+    FILE* get_file() {
+        return file_;
+    }
```

Comment:
`fopen` / `get_file` methods should be removed.

We need just cleanup scope guard (in C++11 it could be `std::unique_ptr`), not a full function wrapper.